contracts/transfer/circuits/src/execute.rs

```diff
@@ -0,0 +1,365 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::{circuit_common_methods, gadgets, rusk_profile_methods};
+
+use crossover::CircuitCrossover;
+use input::{CircuitInput, WitnessInput};
+use output::{CircuitOutput, WitnessOutput};
+
+use anyhow::{anyhow, Result};
+use dusk_plonk::bls12_381::BlsScalar;
+use dusk_plonk::jubjub::JubJubExtended;
+use phoenix_core::Note;
+use poseidon252::sponge;
+use poseidon252::tree::{self, PoseidonBranch};
+use rand_core::{CryptoRng, RngCore};
+use schnorr::gadgets as schnorr_gadgets;
+
+use dusk_plonk::prelude::*;
+
+mod crossover;
+mod input;
+mod output;
+
+#[cfg(test)]
+mod tests;
+
+/// The circuit responsible for creating a zero-knowledge proof
+/// for a 'send to contract transparent' transaction.
+#[derive(Debug, Default, Clone)]
+pub struct ExecuteCircuit<const DEPTH: usize, const CAPACITY: usize> {
+    pi_positions: Vec<PublicInput>,
+    pub inputs: Vec<CircuitInput<DEPTH>>,
+    pub crossover: CircuitCrossover,
+    pub outputs: Vec<CircuitOutput>,
+    pub tx_hash: BlsScalar,
+}
```

Comment:
We can't trivially have iterators in that case because we have variable number of inputs x outputs.

This way, the iterator would be built over an `enum` that covers all the circuit permutations, and this is increased complexity

Reply:
I think it would make the API a way simpler to use. But as @ZER0 said, we can file an issue and discuss/address this in the future.